src/include/mpi.h.in

```diff
@@ -520,7 +520,7 @@ typedef int MPI_Info;
 #define MPI_INFO_NULL         ((MPI_Info)0x1c000000)
 #define MPI_INFO_ENV          ((MPI_Info)0x5c000001)
 #define MPI_MAX_INFO_KEY       255
-#define MPI_MAX_INFO_VAL      1024
+#define MPI_MAX_INFO_VAL     16384
```

Comment:
This is an ABI break. No way.

Reply:
@minsii Any ideas?

Reply:
Then we have to separate info values or find a way to compress.

Reply:
I am thinking that we could allow user to arbitrarily split any hints with a semantics like 
```
key_name-1: [first portion of value]
key_name-2: [second portion of value]
key_name-3: ...
...
```
What do you guys think? @raffenet @minsii

Reply:
@hzhou sorry for the design change. But I found that the new hints are not good solution for the atomicity issue. We need `max_count` information in advance only for `accumulate` and `get_accumulate`. Because for fop/cswap, it is guaranteed that the count is 1 on all processes. 

By default, `accumulate_noncontig_dtype` is true, meaning the user may use noncontig datatype with `accumulate` and `get_accumulate`. We don't want to have native implementation for such case, thus we have to disable all native atomics.

But if the user specifies `accumulate_noncontig_dtype=false`, we need handle only contiguous acc/get_acc (may be arbitrarily long) and fop/cswap with count=1. The only way may break native in our current implementation is that the acc/get_acc data is longer than the hw limit. Rather than asking user to promise that they never use such long data, I think it might make more sense for us to **chunk the long continuous data and send each chunk in native path**. Thus, we can ensure that all atomics are through native path. 

I hope the implementation for chunking long contiguous data will not make the code very complex. @raffenet do you think the change is acceptable for the release?

Reply:
Or, perhaps we should not rush the fix. The fix is not a part of OpenSHMEM FY20 milestone. Only the progress opt is mandatory.

Reply:
Yes, not rushing this part sounds good to me.

Reply:
As an addendum, we had previously floated the idea of breaking ABI for MPICH 4.0, so this type of change might be OK in that timeframe. I need to start an issue to accumulate all the different ABI breakages we want to consider.